functions.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+require_once 'dbConnect.php';
+
+/**
+ * function to call a var dump with pre tags
+ * @param $variabletoDump - the variable to be dumped
+ */
+function var_dumpPre($variabletoDump) {
+    echo '<pre>';
+    var_dump($variabletoDump);
+    echo '</pre>';
+}
+
+/**
+ * get the collection data from the db
+ * @param $db - the database connection
+ * @return mixed - the data from the database in a assoc array
+ */
+function fetchData($db) {
+    $query = $db->prepare('SELECT `id`, `series`, `introducedYR`, `topSpeedKMH`, `topSpeedMPH`, `withdrawn`, `withdrawnYR`, `imgURL` FROM `shinkansens`;');
+    $query->execute();
+    return $query->fetchAll();
+}
+
+/**
+ * function to build the display box for a collection item
+ * @param array $shinkansen - array for one row from the db
+ * @return string - the html to display
+ */
+function collectionBox(array $shinkansen): string {
+    if (!array_key_exists('series', $shinkansen) || (!array_key_exists('imgURL', $shinkansen)) || (!array_key_exists('introducedYR', $shinkansen))) {
+        return 'error! missing expected array key(s): function collectionBox';
+    } elseif (!array_key_exists('topSpeedKMH', $shinkansen) || (!array_key_exists('topSpeedMPH', $shinkansen)) || (!array_key_exists('withdrawn', $shinkansen))) {
+        return 'error! missing expected array key(s): function collectionBox';
+    } elseif (!array_key_exists('withdrawnYR', $shinkansen)) {
+        return 'error! missing expected array key(s): function collectionBox';
+    }
+    $shinkansenBox = '<article><div class=\'item\'>';
+    $shinkansenBox .= '<h2>Series ' . $shinkansen['series'] . '</h2>';
+    $shinkansenBox .= '<img src=\'' . $shinkansen['imgURL'] . '\' alt=\'Series ' . $shinkansen['series'] . ' Bullet Train\' />';
+    $shinkansenBox .= '<ul>';
+    $shinkansenBox .= '<li><span>Introduced:</span> ' . $shinkansen['introducedYR'] . '</li>';
+    $shinkansenBox .= '<li><span>Top speed:</span> ' . $shinkansen['topSpeedKMH'] . 'km/h (' . $shinkansen['topSpeedMPH'] . 'mph)</li>';
+    if ($shinkansen['withdrawn']==1) {
```

Comment:
Could this if statement be a little neater as a ternary statement? would only need to target the content of the li